common/test/acceptance/tests/lms/test_lms.py

```diff
@@ -290,9 +295,17 @@ def test_deferred_verification_enrollment(self):
         enrollment_mode = self.dashboard_page.get_enrollment_mode(self.course_info["display_name"])
         self.assertEqual(enrollment_mode, 'verified')
 
+        # Expect enrollment activated event
+        assert_event_emitted_num_times(
+            self.event_collection,
+            "edx.course.enrollment.activated",
```

Comment:
I may be mistaken, but I think this event is emitted immediately after the user visits the track selection page, when they are enrolled in the course's honor mode. Completing enrollment in the verified mode then triggers a `mode_changed` event.

If we just want to confirm that these events were emitted at some point during the enrollment or upgrade process, putting these checks at the ends of test cases is fine. However, if we're trying to verify the order and timing of these events, I think it'd be good to place these checks that events were emitted in places that more closely reflect the actual behavior.